Classes/Outlet/OutletArgument.php

```diff
@@ -0,0 +1,263 @@
+     * @api
+     */
+    public function getValidators()
+    {
+        return $this->validators;
+    }
+
+    /**
+     * @param string $type
+     * @param array $options
+     */
+    public function addValidator($type, $options) {
+        if (!is_array($options)) {
+            $options = array();
+        }
+        $validator = $this->validatorResolver->createValidator($type, $options);
+        if ($validator === NULL) {
+            throw new NoSuchValidatorException('Unknown Validator: ' . $type, 1478559461 );
+        }
+        $this->validators[] = $validator;
+    }
+
+    /**
+     * Sets the value of this argument.
+     *
+     * @param mixed $rawValue The value of this argument
+     *
+     * @return \TYPO3\CMS\Extbase\Mvc\Controller\Argument
+     * @throws \TYPO3\CMS\Extbase\Property\Exception
+     */
+    public function setValue($rawValue)
+    {
+        $this->value = $this->propertyMapper->convert($rawValue, $this->dataType, $this->propertyMappingConfiguration);
+
+        $this->validationResults = $this->propertyMapper->getMessages();
+        if (count($this->validators) > 0) {
+            $conjunctionValidator = new ConjunctionValidator();
+            foreach ($this->validators as $validator) {
+                $conjunctionValidator->addValidator($validator);
+            }
+            $validationMessages = $conjunctionValidator->validate($this->value);
+            $this->validationResults->merge($validationMessages);
+        }
+        return $this;
+    }
+
+    /**
+     * Returns the value of this argument
+     *
+     * @return mixed The value of this argument - if none was set, NULL is returned
+     * @api
+     */
+    public function getValue()
+    {
+        if ($this->value === null) {
+            return $this->defaultValue;
+        } else {
+            return $this->value;
+        }
+    }
+
+    /**
+     * Return the Property Mapping Configuration used for this argument; can be used by the initialize*action to modify the Property Mapping.
+     *
+     * @return \TYPO3\CMS\Extbase\Mvc\Controller\MvcPropertyMappingConfiguration
+     * @api
+     */
+    public function getPropertyMappingConfiguration()
+    {
+        return $this->propertyMappingConfiguration;
+    }
+
+    /**
+     * @return bool TRUE if the argument is valid, FALSE otherwise
+     * @api
+     */
+    public function isValid()
+    {
+        if ($this->validationResults === NULL) {
+            return TRUE;
+        }
+        return !$this->validationResults->hasErrors();
+    }
+
+    /**
+     * @return \TYPO3\CMS\Extbase\Error\Result Validation errors which have occurred.
+     * @api
+     */
+    public function getValidationResults()
+    {
+        return $this->validationResults;
+    }
+
+    /**
+     * Returns a string representation of this argument's value
+     *
+     * @return string
+     * @api
+     */
+    public function __toString()
```

Comment:
I would drop this string overload - we have `getValue` so we can do `{argument.value}` just fine :)